### PR TITLE
Update caching.md

### LINKED ISSUE #3933: Caching strategies for monorepos (Lerna, Yarn workspaces)
# Missing Information
I was looking for caching techniques for monorepos and didn't see any.

I was in process of converting from bunch of regular single package repos which can use the suggested cache key of `v1-npm-deps-{{ checksum "package-lock.json" }}`; however, for monorepos where there are multiple / nested `package.json`s this does not suffice.

The outer package wouldn't capture changes to inner packages.

The solution I came up with was to simply concatenate checksums of multiple package jsons such as:
`v1-deps-{{ .Branch }}-{{ checksum "package.json" }}-{{ checksum "./packages/models/package.json" }}-{{ checksum "./packages/sdk/package.json" }}-{{ checksum "./packages/ui/package.json" }}-{{ checksum "./packages/samples/package.json" }}`

Example:
`v1-deps-master-G+CP6jVyNkkNpKOWJ4mGVti55LeiwFsAuK3KvWJM5hs=-ae0DuAPUQBWmNGYWamxK282FaJoc_TEFiGp+alY_W2c=-nM+E_1mrETTdBUc63QmS+RM_U6tTssoCjrvSC6Xo9jk=-81coRH8sTogaOf351TSoxPKMoaeUMfm26ID9PQHOjqo=-ya0WvUSus9OaXzcjK1pYuf6NDQUpNlpjij2GrBNoH08=`

However, this doesn't scale well as it would require manual modification everytime a package was added.  I think there would a way to programmatically generate a key string based on package.json's in the repo but that would be more complicated.

# Already Looked
https://circleci.com/docs/2.0/caching/#restoring-cache

# Potential Locations
https://circleci.com/docs/2.0/caching/#restoring-cache

I expect it to be found on this page. I understand my use case is more rare than traditional but it wouldn't hurt to add it as a small section at the bottom which could be linked too.

### PR DESCRIPTION
**Pull request by thomasr8**
Added customer caching example for monorepos in Restoring Cache section. This fixes #3933

# Description
A brief description of the changes.

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

### CODE REVIEW

File: jekyll/_cci2/caching.md [hunk 1]
```diff
@@ -104,6 +104,45 @@ Here, the first key concatenates the checksum of `package-lock.json` file into t
 
 The next key does not have a dynamic component to it, it simply is a static string: `v1-npm-deps-`. If you would like to invalidate your cache manually, you can bump `v1` to `v2` in your `config.yml` file. In this case, you would now have a new cache key `v2-npm-deps`, which will trigger the storing of a new cache.
 
+### Using Caching in Monorepos
+
+There are many different approaches to utilizing caching in monorepos. The customer-provided example below shows how a user can create and then build a concatenated `package-lock` file that can be used as a checksum for the caching key.
```
Comment (glenjamin) [inline jekyll/_cci2/caching.md]: Suggestion to add: this sort of approach can be used whenever you need to managed a shared cache based on multiple files in different parts of your monorepo

File: jekyll/_cci2/caching.md [hunk 2]
```diff
@@ -104,6 +104,45 @@ Here, the first key concatenates the checksum of `package-lock.json` file into t
...
+
+2) Use custom command in build to generate the concatenated `package-lock` file
+
+{% raw %}
+```yaml
+    steps:
+        - checkout
+        - create_concatenated_package_lock:
+          filename: combined-package-lock.txt
+    Use combined-package-lock.text in cache key
```
Comment (glenjamin) [inline jekyll/_cci2/caching.md]: I think this should be a comment

File: jekyll/_cci2/caching.md (hunk 1 above)
Comment (thomasr8) [reply jekyll/_cci2/caching.md]: fixed

File: jekyll/_cci2/caching.md (hunk 2 above)
Comment (thomasr8) [reply jekyll/_cci2/caching.md]: fixed

**Pull request merged**